job-service/src/main/java/com/hpe/caf/services/job/api/filter/JobFilterQuery.java

```diff
@@ -0,0 +1,163 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.services.job.api.filter;
+
+import com.hpe.caf.services.job.exceptions.FilterException;
+import com.healthmarketscience.sqlbuilder.BinaryCondition;
+import com.healthmarketscience.sqlbuilder.ComboCondition;
+import com.healthmarketscience.sqlbuilder.CommonTableExpression;
+import com.healthmarketscience.sqlbuilder.InCondition;
+import com.healthmarketscience.sqlbuilder.NotCondition;
+import com.healthmarketscience.sqlbuilder.Condition;
+import com.healthmarketscience.sqlbuilder.SelectQuery;
+import com.healthmarketscience.sqlbuilder.UnaryCondition;
+import com.healthmarketscience.sqlbuilder.dbspec.Table;
+import com.healthmarketscience.sqlbuilder.dbspec.basic.DbColumn;
+import com.healthmarketscience.sqlbuilder.dbspec.basic.DbSchema;
+import com.healthmarketscience.sqlbuilder.dbspec.basic.DbSpec;
+import com.healthmarketscience.sqlbuilder.dbspec.basic.DbTable;
+import cz.jirutka.rsql.parser.ast.ComparisonOperator;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class JobFilterQuery
+{
+    private static final Map<String, DbColumn> COLUMN_MAPPINGS;
+    private static final DbTable JOB_TABLE;
+    private static final DbTable LABEL_TABLE;
+
+    private static final DbColumn JOB_ID;
+    private static final DbColumn JOB_NAME;
+    private static final DbColumn JOB_PARTITION_ID;
+    private static final DbColumn JOB_CREATE_TIME;
+    private static final DbColumn JOB_LAST_MODIFIED;
+    private static final DbColumn JOB_STATUS;
+    private static final DbColumn JOB_PERCENTAGE_COMPLETE;
+
+    private static final DbColumn LABEL_JOB_ID;
+    private static final DbColumn LABEL_LABEL;
+    private static final DbColumn LABEL_PARTITION_ID;
+    private static final DbColumn LABEL_VALUE;
+
+    static {
+        JOB_TABLE = new DbTable(new DbSchema(new DbSpec(), ""), "job", "job");
+        LABEL_TABLE = new DbTable(new DbSchema(new DbSpec(), ""), "label", "label");
+
+        JOB_ID = JOB_TABLE.addColumn("job_id");
+        JOB_NAME = JOB_TABLE.addColumn("name");
+        JOB_PARTITION_ID = JOB_TABLE.addColumn("partition_id");
+        JOB_CREATE_TIME = JOB_TABLE.addColumn("create_date");
+        JOB_LAST_MODIFIED = JOB_TABLE.addColumn("last_update_date");
+        JOB_STATUS = JOB_TABLE.addColumn("job_status");
+        JOB_PERCENTAGE_COMPLETE = JOB_TABLE.addColumn("percentage_complete");
+
+        LABEL_PARTITION_ID = LABEL_TABLE.addColumn("partition_id");
+        LABEL_LABEL = LABEL_TABLE.addColumn("label");
+        LABEL_JOB_ID = LABEL_TABLE.addColumn("job_id");
+        LABEL_VALUE = LABEL_TABLE.addColumn("value");
+
+        COLUMN_MAPPINGS = new ConcurrentHashMap<>();

```

Comment:
You can just use an ordinary `HashMap` here.  You only need a `ConcurrentHashMap` if there was a possibility of multiple threads modifying the map concurrently, but in this case the map is never modified after it is initialized.